src/main/java/com/ebay/myriad/configuration/MyriadConfiguration.java

```diff
@@ -123,11 +123,35 @@
     @NotEmpty
     private Map<String, String> yarnEnvironment;
 
+
+    public MyriadConfiguration(){}
+
+    /*
+        Copy constructor used by rest api to map this config to GetConfigurationResponse
+     */
+    protected MyriadConfiguration(MyriadConfiguration other) {
+        this.mesosMaster = other.getMesosMaster();
+        this.checkpoint = other.isCheckpoint();
+        this.frameworkFailoverTimeout = other.getFrameworkFailoverTimeout();
+        this.frameworkName = other.getFrameworkName();
+        this.frameworkRole = other.getFrameworkRole();
+        this.profiles = other.getProfiles();
+        this.rebalancer = other.isRebalancer();
+        this.nodemanager = other.getNodeManagerConfiguration();
+        this.executor = other.getMyriadExecutorConfiguration();
+        this.nativeLibrary = other.getNativeLibrary();
+        this.zkServers = other.getZkServers();
+        this.zkTimeout = other.getZkTimeout();
+        this.restApiPort = other.getRestApiPort();
+        this.yarnEnvironment = other.getYarnEnvironment();
+    }
```

Comment:
Please also include mesosAuthenticationPrincipal and mesosAuthenticationSecretFilename and anything else added in the meantime.